### PR TITLE
Base implementation with fetch/push functionality

### PR DESCRIPTION
**Pull request by Qinusty**
Functionality:
- Push service supporting blobs and directories with qualifiers
- Fetch service supporting:
  - Pushed blobs/Directories with exact qualifier/uri matching
  - Fetching blobs from URIs via http (and pushing to a GRPC CAS)
    - Support for checksum.sri qualifier

Tested with buildstream ([remote-asset](https://gitlab.com/BuildStream/buildstream/-/tree/juerg/remote-asset)) / bazel (3.3.1)

Related issues: #1, #2, #3

### CODE REVIEW

**Comment by tomcoldrick-ct** [comment]
We should remove the branch CI at some point, but I think that can wait until this is merged.

**Review by tomcoldrick-ct — APPROVED** [review]
One nit we might want to address is tidying up the go imports, I think the other bb-* repos split local imports into a separate block.

**Comment by Qinusty** [comment]
> We should remove the branch CI at some point, but I think that can wait until this is merged.

Agreed.

> One nit we might want to address is tidying up the go imports, I think the other bb-* repos split local imports into a separate block.

Agreed. We can address this in a follow up PR.

**Pull request merged**
